src/pathfinding_service/database.py

```diff
@@ -206,31 +208,84 @@ def get_token_networks(self) -> Iterator[TokenNetwork]:
         for row in self.conn.execute("SELECT address FROM token_network"):
             yield TokenNetwork(token_network_address=decode_hex(row[0]))
 
-    def insert_feedback_token(self, token: FeedbackToken) -> None:
+    def prepare_feedback(self, token: FeedbackToken, route: List[Address]) -> None:
+        hexed_route = [to_checksum_address(e) for e in route]
         token_dict = dict(
             token_id=token.id.hex,
             creation_time=token.creation_time,
             token_network_address=to_checksum_address(token.token_network_address),
+            route=json.dumps(hexed_route),
         )
         self.conn.execute(
             """
-            INSERT INTO feedback_token (
-                token_id, creation_time, token_network_address
+            INSERT INTO feedback (
+                token_id, creation_time, token_network_address, route
             ) VALUES (
                 :token_id,
                 :creation_time,
-                :token_network_address
+                :token_network_address,
+                :route
             )
         """,
             token_dict,
         )
 
+    def has_feedback_for(self, token: FeedbackToken, route: List[Address]) -> bool:
+        hexed_route = [to_checksum_address(e) for e in route]
+        feedback = self.conn.execute(
+            """SELECT successful FROM feedback WHERE
+                token_id = ? AND
+                token_network_address = ? AND
+                route = ?;
+            """,
+            [
+                token.id.hex,
+                to_checksum_address(token.token_network_address),
+                json.dumps(hexed_route),
+            ],
+        ).fetchone()
+
+        if feedback:
+            return feedback["successful"] is not None
```

Comment:
You can also do `SELECT successful IS NOT NULL` right in the query. To me that is more straighforward, but I'm also used to have logic in SQL.